components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java

```diff
@@ -381,21 +394,31 @@ public static OAuthApplicationInfo retrieveOAuthApplication(String consumerKey)
-            APIDefinition definitionFromSwagger20 = new APIDefinitionFromSwagger20();
-            API api = new API(new APIIdentifier(RestApiConstants.REST_API_PROVIDER, RestApiConstants.REST_API_STORE_CONTEXT,
-                    RestApiConstants.REST_API_STORE_VERSION));
-            //Get URL templates from swagger content we created
-            try {
-                publisherResourceMappings = definitionFromSwagger20.getURITemplates(api, sb.toString());
+            */
+
+                APIDefinition definitionFromSwagger20 = new APIDefinitionFromSwagger20();
+                //Get URL templates from swagger content we created
+                publisherResourceMappings = definitionFromSwagger20.getURITemplates(api, definition);
             } catch (APIManagementException e) {
                 log.error("Error while reading resource mappings for API: " + api.getId().getApiName());
+            } catch (IOException e) {
+                log.error("Error while reading the swagger definition for API: " + api.getId().getApiName());
```

Comment:
Fixed in #1494(https://github.com/wso2/carbon-apimgt/commit/d73b3e2744cf2b38b48b6902a2e67b92590df26e)